runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaMergedFactory.java

```diff
@@ -204,42 +204,37 @@ public MessageConsumer newStream(
 
         MessageConsumer newStream = null;
 
-        final KafkaBindingConfig binding = supplyBinding.apply(routeId);
-
-        if (binding != null && binding.merged(topicName))
-        {
-            final long resolvedId = routeId;
-            final ArrayFW<KafkaOffsetFW> partitions = kafkaMergedBeginEx.partitions();
-
-            final KafkaOffsetFW partition = partitions.matchFirst(p -> p.partitionId() == -1L);
-            final long defaultOffset = partition != null ? partition.partitionOffset() : HISTORICAL.value();
-
-            final Long2LongHashMap initialOffsetsById = new Long2LongHashMap(-3L);
-            partitions.forEach(p ->
-            {
-                final long partitionId = p.partitionId();
-                if (partitionId >= 0L)
-                {
-                    final long partitionOffset = p.partitionOffset();
-                    initialOffsetsById.put(partitionId, partitionOffset);
-                }
-            });
-
-            newStream = new KafkaMergedStream(
-                    sender,
-                    routeId,
-                    initialId,
-                    affinity,
-                    authorization,
-                    topicName,
-                    resolvedId,
-                    capabilities,
-                    initialOffsetsById,
-                    defaultOffset,
-                    isolation,
-                    deltaType,
-                    ackMode)::onMergedMessage;
-        }
```

Comment:
This change loses the `binding != null` condition too.

Reply:
`binding` is not getting used at any other place, so I wasn’t sure if a null check is required.

Reply:
done, reverted the `binding != null` condition.